source/common/router/config_impl.cc

```diff
@@ -65,6 +67,82 @@ Optional<uint64_t> HashPolicyImpl::generateHash(const Http::HeaderMap& headers)
   return hash;
 }
 
+HeaderFormatterSharedPtr RequestHeaderParser::parseInternal(const std::string& format) {
```

Comment:
Can we please move all of this code into a new .cc/.h file. This is pretty self contained and this file is getting too huge as it is.